src/gui/Preferences.cc

```diff
@@ -1403,3 +1398,12 @@ void Preferences::updateGUIFontSize(QComboBox *fsSelector, const QString &settin
     BlockSignals<QComboBox *>(fsSelector)->setEditText(fontsize);
   }
 }
+
+Preferences* GlobalPreferences::inst()
+{
+    static Preferences* instance {nullptr};
+    if(instance==nullptr){
+        instance = new Preferences();
+    }
```

Comment:
You can rewrite this to `static auto *instance = new Preferences();`

Reply:
Your proposition makes me refresh my memory about how static block variables are initialized. And from my understanding initialization would be done the first time the block enters.
So I wonder if 
 ```cpp
Preferences* GlobalPreferences::inst()
{
    static Preferences instance;
    return &instance;  
}
```
Isn't preferable ?

Reply:
Yeah that should work too, perhaps slightly preferable, but I don't have a strong opinion right now.

Reply:
I think that the version without pointer guarantees that the Preferences destructor is called on exit.